jdaviz/core/template_mixin.py

```diff
@@ -2108,3 +2112,152 @@ def unmix_state(self):
+
+    @property
+    def display_name(self):
+        return f'{self._plugin._plugin_name}: {self._component_type}'
+
+    def vue_popout(self, data=None):
+        self.show(loc='popout')
+
+    def show(self, loc="inline", title=None):  # pragma: no cover
+        """Display the component UI.
+
+        Parameters
+        ----------
+        loc : str
+            The display location determines where to present the viz app.
+            Supported locations:
+
+            "inline": Display the component inline in a notebook.
+
+            "app": Display below the viewers in the main app.
+
+            "sidecar": Display the component in a separate JupyterLab window from the
+            notebook, the location of which is decided by the 'anchor.' right is the default
+
+                Other anchors:
+
+                * ``sidecar:right`` (The default, opens a tab to the right of display)
+                * ``sidecar:tab-before`` (Full-width tab before the current notebook)
+                * ``sidecar:tab-after`` (Full-width tab after the current notebook)
+                * ``sidecar:split-right`` (Split-tab in the same window right of the notebook)
+                * ``sidecar:split-left`` (Split-tab in the same window left of the notebook)
+                * ``sidecar:split-top`` (Split-tab in the same window above the notebook)
+                * ``sidecar:split-bottom`` (Split-tab in the same window below the notebook)
+
+                See `jupyterlab-sidecar <https://github.com/jupyter-widgets/jupyterlab-sidecar>`_
+                for the most up-to-date options.
+
+            "popout": Display the component in a detached display. By default, a new
+            window will open. Browser popup permissions required.
+
+                Other anchors:
+
+                * ``popout:window`` (The default, opens Jdaviz in a new, detached popout)
+                * ``popout:tab`` (Opens Jdaviz in a new, detached tab in your browser)
+
+        title : str, optional
+            The title of the sidecar tab.  Defaults to the name of the component.
+
+            NOTE: Only applicable to a "sidecar" display.
+
+        Notes
+        -----
+        If "sidecar" is requested in the "classic" Jupyter notebook, the component will appear
+        inline, as only JupyterLab has a mechanism to have multiple tabs.
+        """
+        title = title if title is not None else self.display_name
+        show_widget(self, loc=loc, title=title)
+
+
+class Table(PluginSubcomponent):
+    """
+    """
+    template_file = __file__, "../components/plugin_table.vue"
+
+    headers_visible = List([]).tag(sync=True)  # list of strings
+    headers_avail = List([]).tag(sync=True)   # list of strings
+    items = List().tag(sync=True)  # list of dictionaries, pass single dict to add_row
+
+    def __init__(self, plugin, *args, **kwargs):
+        self._qtable = None
+        super().__init__(plugin, 'Table', *args, **kwargs)
+
+    def add_row(self, item):
```

Comment:
Using the same method name as `astropy.table.QTable` but have the behavior to be different is very confusing to me. Maybe change to `add_row_or_replace` or consider refactoring.

Reply:
it does effectively add a row to the underlying `QTable` object though, it just also manages making sure it can be synced with the UI table.  I personally think its useful to have the same method name and would be hesitant to make a more verbose name just because `QTable` already uses this.  But if the consensus is that this will be confusing, perhaps `add_item` would work?

Reply:
`add_item` would confuse me less than `add_row`, yes.